src/indexed_merkle_tree/mod.rs

```diff
@@ -278,22 +278,23 @@ impl IndexedMerkleTree {
                     Some(label)
                 }
             };
+            //TODO: DRY
             ordered_derived_dict_keys
                 .iter()
                 .enumerate() // use index 
                 .find(|(_, k)| {
-                    *k == &label.clone().unwrap() // ohne dereferenzierung wird ein &&String mit &String verglichen
+                    *k == &label.clone().unwrap() // without dereferencing a &&String is compared with &String
                 })
                 .unwrap()
-                .0 // enumerate gibt tupel zurück, also index zurückgeben
+                .0 // enumerate returns tuple, so we have to return index
         });
     
         // Add empty nodes to ensure the total number of nodes is a power of two.
         while nodes.len() < next_power_of_two {
             nodes.push(Node::initialize_leaf(false, true, Node::EMPTY_HASH.to_string(), Node::EMPTY_HASH.to_string(), Node::TAIL.to_string()));
         }
     
-        // baum erstellen und dabei alle nodes überprüfen, ob sie linkes oder rechtes kind sind
+        // create tree checking all nodes if they are left or right child
```

Comment:
i dont understand this comment, what do you mean if they are left or right child?